students/esokada/lesson03/mailroom.py

```diff
@@ -15,24 +15,22 @@
 #functions
 
 def send_thankyou():
-    donorname = input("Please input a name:\n")
-    if donorname == "cancel":
-        return
-    if donorname == "list":
-        for i in donors:
-            print(i[0])
+    while True:
         donorname = input("Please input a name:\n")
-        if donorname == "cancel":
+        if donorname == "C":
             return
+        if donorname == "list":
+            for i in donors:
+                print(i[0])
+        else:
+            break
     donationamount = input("Please enter a donation amount:\n")
-    if donationamount == "cancel":
+    if donationamount == "C":
         return
-    existingdonor = False
     for d in donors:
         if d[0] == donorname:
-            existingdonor = True
             break
-    if existingdonor == False:
+    else:
```

Comment:
this is close.. but here is a more condensed version:
```
    while True:
        donorname = input("Please input a name, "list" command or "C" to cancel:\n")  # clarify what a user can input
        if donorname == "C":
            return
        elif donorname == "list":
            for donor in donors:
                print(donor[0])
        else:
            # here we will be adding donors
            donation_amount = input("Please enter a donation amount:\n")
            for donor in donors:
                if donor[0] == donorname:
                    donor[1].append(donation_amount)
            else:
                donors.append([donorname, [donation_amount]])

            print(f"Dear {donorname}, thank you for your generous donation of ${float(donation_amount):.2f}.\n")
```

Reply:
please review above ^^ let me know if you have questions.

Reply:
Thanks, that makes sense!  I will incorporate these changes into next mailroom assignment.

Reply:
sounds good!